### PR TITLE
nm: update page

### PR DESCRIPTION
**Pull request by EmilySeville7cfg**
- [ ] The page (if new), does not already exist in the repository.
- [x] The page is in the correct platform directory (`common/`, `linux/`, etc.)
- [x] The page has 8 or fewer examples.
- [x] The PR title conforms to the recommended [templates](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#commit-message).
- [x] The page follows the [content guidelines](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#guidelines).
- [x] The page description includes a link to documentation or a homepage (if applicable).

**Version of the command being documented (if known):**

## ToDo

- [x] prefer long options
- [x] fix placeholders

### CODE REVIEW

**Review by navarroaxel — APPROVED** [review]

File: pages/common/nm.md [hunk 1]
```diff
@@ -1,19 +1,20 @@
...
 
 - List global (extern) functions in a file (prefixed with T):
 
-`nm -g {{file.o}}`
+`nm --extern-only {{path/to/file.o}}`
 
 - List only undefined symbols in a file:
 
-`nm -u {{file.o}}`
+`nm --undefined-only {{path/to/file.o}}`
```
Comment (pixelcmtd) [inline pages/common/nm.md:12]:
```suggestion
`nm -u {{path/to/file.o}}`
```

File: pages/common/nm.md [hunk 2]
```diff
@@ -1,19 +1,20 @@
...
 
 - List all symbols, even debugging symbols:
 
-`nm -a {{file.o}}`
+`nm --debug-syms {{path/to/file.o}}`
```
Comment (pixelcmtd) [inline pages/common/nm.md:16]:
```suggestion
`nm -a {{path/to/file.o}}`
```

**Review by pixelcmtd — CHANGES_REQUESTED** [review]

**Review by mfrw — COMMENTED** [review]
Given these pages are in common, should we create new pages for `osx` which in my understanding would not have the long version of the flags ?

**Comment by pixelcmtd** [comment]
> should we create new pages for `osx` which [...] would not have the long version of the flags ?

if we were to do something like that, i'd instead make new `linux` pages which have the long options, because initially the long options were a gnu-specific thing, for example, early freebsd versions didn't have it either.

**Comment by EmilySeville7cfg** [comment]
> if we were to do something like that, i'd instead make new linux pages which have the long options, because initially the long options were a gnu-specific thing, for example, early freebsd versions didn't have it either.

I think we can use `general/` for pages with most portable options (short/long) and `linux/` for just GNU options.

**Pull request merged**